Kitodo/src/main/java/org/kitodo/services/data/TemplateService.java

```diff
@@ -73,19 +75,41 @@ public void saveToIndex(Template template) throws CustomResponseException, IOExc
     }
 
     /**
-     * Method saves process and properties related to modified template.
+     * Method manages process and properties related to modified template.
      *
      * @param template
      *            object
      */
     protected void manageDependenciesForIndex(Template template)
             throws CustomResponseException, DataException, IOException {
-        if (template.getProcess() != null) {
-            serviceManager.getProcessService().saveToIndex(template.getProcess());
+        manageProcessDependenciesForIndex(template);
+        managePropertiesDependenciesForIndex(template);
+    }
+
+    private void manageProcessDependenciesForIndex(Template template) throws CustomResponseException, IOException {
+        Process process = template.getProcess();
+        if (template.getIndexAction() == IndexAction.DELETE) {
+            if (process != null) {
```

Comment:
A template should never exists without a corresponding process. It could be possible that a process has no template.